native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ArgumentHandler.java

```diff
@@ -88,20 +92,37 @@ public class ArgumentHandler {
     private static final int T_BOOLEAN = 5;
     private static final int T_INPUT_OBJECT = 22;
     private static final int T_LIST = 23;
+    private static final ArrayList<String> idsList = new ArrayList<>();
+    private static final String ID_ANNOTATION = "ID";
 
     public ArgumentHandler(MethodType method, BObject context, BObject field, BObject responseGenerator,
                            boolean validation) {
         this.method = method;
         this.fileInfo = (BMap<BString, Object>) context.getNativeData(FILE_INFO_FIELD);
         this.context = context;
         this.field = field;
+        getIdTypeArguments(method.getAnnotations());
         this.argumentsMap = ValueCreator.createMapValue();
         this.responseGenerator = responseGenerator;
         this.validation = validation;
         BObject fieldNode = this.field.getObjectValue(INTERNAL_NODE);
         this.populateArgumentsMap(fieldNode);
     }
 
+    private void getIdTypeArguments(BMap<BString, Object> annotations) {
```

Comment:
Since we are not returning any value here maybe we could change the method name to `populateIdTypeArguments` or `findIdTypeArguments`

Reply:
+1 to `populateIdTypeArguments`